src/authgss_hash.c

```diff
@@ -324,3 +372,39 @@ void authgss_ctx_gc_idle(void)
 	/* perturb by 1 */
 	(void)IDLE_NEXT();
 }
+
+void authgss_ctx_hash_clear(void)
+{
+	struct rbtree_x_part *xp;
+	struct authgss_x_part *axp;
+	struct svc_rpc_gss_data *gd;
+	int ix;
+
+	authgss_hash_init();
+
+	for (ix = 0; ix < authgss_hash_st.xt.npart; ++ix) {
+		xp = &(authgss_hash_st.xt.tree[ix]);
+		axp = (struct authgss_x_part *)xp->u1;
+		mutex_lock(&xp->mtx);
+
+ again:
```

Comment:
My earlier implementation was an effort to be consistent with the existing hash iteration in the `authgss_ctx_gc_idle` method.
However, I agree with your opinion to avoid using labels where possible. I have updated the PR to now iterate over the tailq.